TrainingApp/Controller/SettingsViewController.swift

```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
         setupTapGestureForViews()
-    }
-    
-    // MARK: - Actions
-    
-    @IBAction private func darkModeTogglePressed(_ sender: UIButton) {
-        if darkModeToggleButton.currentImage == R.image.button() {
-            darkModeIsOn()
-        }
-        else {
-            darkModeIsOff()
+        checkThemeSwitchState()
+        checkNotificationSwitchState()
+        print(FLIGHT_KEY)
+        print(EXECUTIVE_KEY)
+        updateTheme()
+        updateNotifications()
+
+        darkModeSwitch.setOnValueChangeListener {
+            if self.darkModeSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_THEME)
+                self.userDefaults.set(self.DARK_THEME, forKey: self.THEME_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_THEME)
+                self.userDefaults.set(self.LIGHT_THEME, forKey: self.THEME_KEY)
+            }
+            self.updateTheme()
         }
-    }
-    
-    @IBAction private func notificationsTogglePressed(_ sender: UIButton) {
-        if notificationsToggleButton.currentImage == R.image.button() {
-            notificationIsOn()
+        
+        notificationsSwitch.setOnValueChangeListener {
+            if self.notificationsSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_NOTIFICATION)
+                self.userDefaults.set(self.NOTIFICATION_ON, forKey: self.NOTIFICATION_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_NOTIFICATION)
+                self.userDefaults.set(self.NOTIFICATION_OFF, forKey: self.NOTIFICATION_KEY)
+            }
+            self.updateNotifications()
         }
-        else {
-            notificationIsOff()
+        
+        flightUpdatesSwitch.setOnValueChangeListener {
+            if self.flightUpdatesSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_FLIGHT)
+                self.userDefaults.set(self.FLIGHT_ON, forKey: self.FLIGHT_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_FLIGHT)
+                self.userDefaults.set(self.FLIGHT_OFF, forKey: self.FLIGHT_KEY)
+            }
         }
-    }
-    
-    @IBAction private func flightUpdatesTogglePressed(_ sender: UIButton) {
-        if flightUpdatesToggle.currentImage == R.image.button() {
-            flightUpdatesIsOn()
+        
+        executiveProgramSwitch.setOnValueChangeListener {
+            if self.executiveProgramSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_EXECUTIVE)
+                self.userDefaults.set(self.EXECUTIVE_ON, forKey: self.EXECUTIVE_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_EXECUTIVE)
+                self.userDefaults.set(self.EXECUTIVE_OFF, forKey: self.EXECUTIVE_KEY)
+            }
```

Comment:
Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

Reply:
Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.